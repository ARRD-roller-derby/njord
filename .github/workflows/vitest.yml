name: Vitest
on: push
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v2
      with:
        node-version: '16.x'
    - name: Install dependencies
      run: npm ci
    - name: Run Vitest tests
      run: npm run coverage
    - uses: actions/upload-artifact@v2
      if: always()
      with:
        name: vitest-report
        path: vitest-report/
        retention-days: 30