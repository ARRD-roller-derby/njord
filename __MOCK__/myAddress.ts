import dayjs from 'dayjs'
import { addressInterface } from '../types/address.interface'
import { addressType } from '../types/addressType.enum'
export const myAddress: addressInterface = {
  _id: 'fake_id',
  label: 'my address',
  lat: 2.222222,
  lon: 3.333333,
  ownerId: 'my_id',
  city: 'Gotham City',
  zipcode: '010101',
  street: 'Rue du crime',
  type: addressType.personnal,
  updatedAt: dayjs('01/01/2022').toDate(),
}
