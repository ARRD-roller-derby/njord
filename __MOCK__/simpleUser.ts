import { UserInterface } from "../types/User.interface";

export const simpleUser: UserInterface = {
  _id: 'simple_user_id',
  email: 'simple@user.fr',
  name: 'simple',
  lastname: 'user',
  derbyName: 'mega shark',
  numRoster: '01',
  msp: false,
  mst: true,
  profiles: [],
  dailyContestAvgTime: 13,
  dailyContestAvgAccuracy: 50,
  league: {
    id: 'my_league_id',
    shortName: 'my league'
  }
}