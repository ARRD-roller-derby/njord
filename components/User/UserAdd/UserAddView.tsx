import classes from './UserAdd.module.css'
import FullscreenModale from '../../_ui/FullscreenModale/FullscreenModale'
import UserAddForm from '../UserAddForm/UserAddForm'

interface props {
  readonly reSync: Function
  readonly defaultValue: string
  readonly handleClose: Function
  readonly handleOpen: Function
  readonly isOpen: boolean
}

export default function UserAddView({
  defaultValue,
  isOpen,
  handleClose,
  handleOpen,
  reSync,
}: props) {
  return (
    <div className={classes.container}>
      {isOpen && (
        <FullscreenModale setClose={handleClose}>
          <UserAddForm
            setClose={handleClose}
            defaultValue={defaultValue}
            reSync={reSync}
          />
        </FullscreenModale>
      )}
      <button onClick={() => handleOpen()}>Ajouter un utilisateur</button>
    </div>
  )
}
