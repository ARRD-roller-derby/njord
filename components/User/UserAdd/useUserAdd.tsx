import { useState } from 'react'
import { UserInterface } from '../../../types/User.interface'

export default function useUserAdd(openPopin: Function) {
  const [isOpen, setIsOpen] = useState<boolean>(false)

  function handleClose(user: UserInterface) {
    setIsOpen(false)
    openPopin(user)
  }

  function handleOpen() {
    setIsOpen(true)
  }

  return {
    isOpen,
    handleClose,
    handleOpen,
  }
}
