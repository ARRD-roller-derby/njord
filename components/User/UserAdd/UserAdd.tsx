import UserAddView from './UserAddView';
import useUserAdd from './useUserAdd'

interface props {
  readonly reSync: Function
  readonly defaultValue: string
  readonly openPopin: Function
}

export default function UserAdd({ defaultValue, openPopin, reSync }: props) {
  const { isOpen, handleClose, handleOpen } = useUserAdd(openPopin)

  return (
    <UserAddView
      defaultValue={defaultValue}
      isOpen={isOpen}
      handleOpen={handleOpen}
      handleClose={handleClose}
      reSync={reSync}
      
    />
  )
}
