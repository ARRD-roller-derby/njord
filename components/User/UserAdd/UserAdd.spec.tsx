import { afterEach, describe, expect, it, vi } from 'vitest'
import { render, cleanup, renderHook, act } from '@testing-library/react'
import UserAdd from './UserAdd'
import useUserAdd from './useUserAdd'
import { simpleUser } from '../../../__MOCK__/simpleUser'

describe('<UserAdd />', () => {
  afterEach(cleanup)

  it('Check snapshot', () => {
    const { asFragment } = render(
      <UserAdd defaultValue="test@test.fr" openPopin={vi.fn} reSync={vi.fn} />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('Test hook functions', () => {
    const triggerOpenPopin = vi.fn(),
      { result } = renderHook(() => useUserAdd(triggerOpenPopin))

    //default state
    expect(result.current.isOpen).toBeFalsy()
    //set isOpen
    act(() => {
      result.current.handleOpen()
    })
    expect(result.current.isOpen).toBeTruthy()
    
    //check if openPopin is called
    act(() => {
      result.current.handleClose(simpleUser)
    })
    expect(triggerOpenPopin).toBeCalled()
    expect(result.current.isOpen).toBeFalsy()
  })
})
