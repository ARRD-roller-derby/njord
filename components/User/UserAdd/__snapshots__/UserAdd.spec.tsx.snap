// Vitest Snapshot v1

exports[`<UserAdd /> > Check snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      Ajouter un utilisateur
    </button>
  </div>
</DocumentFragment>
`;
